### PR TITLE
fixed deprecation warnings in swift 5

### PR DESCRIPTION
**Pull request by odyth**
this fixes https://github.com/daltoniam/Starscream/issues/640

### CODE REVIEW

**Comment by WyattMufson** [comment]
@fassko any idea when this will get merged in?

File: Sources/Starscream/Compression.swift [hunk 1]
```diff
@@ -56,8 +56,9 @@ class Decompressor {
     }
 
     func decompress(_ data: Data, finish: Bool) throws -> Data {
-        return try data.withUnsafeBytes { (bytes:UnsafePointer<UInt8>) -> Data in
-            return try decompress(bytes: bytes, count: data.count, finish: finish)
+        return try data.withUnsafeBytes { (bytes:UnsafeRawBufferPointer) -> Data in
```
Comment (fassko) [inline Sources/Starscream/Compression.swift]: I don't want to nit pick but could we add space after `bytes:`

File: Sources/Starscream/WebSocket.swift [hunk 2]
```diff
@@ -279,8 +279,9 @@ open class FoundationStream : NSObject, WSStream, StreamDelegate  {
             var peerNameLen: Int = 0
             SSLGetPeerDomainNameLength(sslContextOut, &peerNameLen)
             var peerName = Data(count: peerNameLen)
-            let _ = peerName.withUnsafeMutableBytes { (peerNamePtr: UnsafeMutablePointer<Int8>) in
-                SSLGetPeerDomainName(sslContextOut, peerNamePtr, &peerNameLen)
+            let _ = peerName.withUnsafeMutableBytes { (peerNamePtr:UnsafeMutableRawBufferPointer) in
```
Comment (fassko) [inline Sources/Starscream/WebSocket.swift]: I don't want to nit pick but could we add space after `bytes:`

File: Sources/Starscream/Compression.swift (hunk 1 above)
Comment (odyth) [reply Sources/Starscream/Compression.swift]: added a space

File: Sources/Starscream/WebSocket.swift (hunk 2 above)
Comment (odyth) [reply Sources/Starscream/WebSocket.swift]: added a space

**Comment by WyattMufson** [comment]
@fassko looks like the changes you've requested have been done

**Comment by acmacalister** [comment]
Thanks for you PR! The codebase has actually under gone a major refactor and should be updated for Swift 5. Feel free to check it out here #653. Feedback welcomed!

**Pull request closed without merging**